plugins/modules/lambda.py

```diff
@@ -808,6 +808,9 @@ def main():
             module.fail_json(msg="Unable to get function information after updating")
         response = format_response(response)
         # We're done
+        # "ZipFile" attribute contains non UTF-8 data. Ansible considers it an error
+        # starting with version 2.18. Removing it from the output avoids the error.
+        code_kwargs.pop("ZipFile", None)
```

Comment:
The `ZipFile` is added [here](https://github.com/ansible-collections/amazon.aws/blob/598a2d12e45dbc5200936a7a2a78ff751e9fd397/plugins/modules/lambda.py#L514). Ideally, we should remove the function that updates this value in the return dictionary. However, could we instead encode it with UTF-8 rather than removing it? Removing it might impact existing usage.

Reply:
We can't encode it as UTF-8 because it's binary data, not text. If we want to continue returning it we need to base64 encode it. However, I'm not sure what the value is in outputing the binary contents of the zipfile, and from what I can see, it's not a documented return value.

Reply:
Thanks for the clarification, @gravesm. If we decide to remove the ZipFile field from the return dictionary, would it be okay also to remove the [function](https://github.com/ansible-collections/amazon.aws/blob/598a2d12e45dbc5200936a7a2a78ff751e9fd397/plugins/modules/lambda.py#L514) and its corresponding call? cc @ichekaldin

Reply:
No, the binary contents are used in the boto call to update the function: https://github.com/ansible-collections/amazon.aws/blob/598a2d12e45dbc5200936a7a2a78ff751e9fd397/plugins/modules/lambda.py#L799

I think what's done in this PR is the right way to handle it. I would consider this a bugfix, not a breaking change, because there's virtually no way this data has ever been usable. For ansible to output this it will have had to drop bytes it can't decode, so what's output has probably never been the actual contents of the zipfile.

Reply:
I missed that line. Thanks!

Reply:
@GomathiselviS Does this PR need to be backported?

Reply:
Yes. It can be. I will add the label.